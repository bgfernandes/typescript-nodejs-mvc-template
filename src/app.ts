import express from 'express';
import config from './config/config';

const app = express();
const port = config.port;

app.get('/', (_req : express.Request, res: express.Response) => {
  res.send('Hello World! Running on ' + config.env);
});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
app.use((error: any, _req: express.Request, res: express.Response, next: express.NextFunction) => {
  console.error(error);

  if (res.headersSent) {
    return next(error);
  }

  res.status(500).send('INTERNAL SERVER ERROR');
});

app.listen(port, () => {
  return console.log(`Server is listening on http://localhost:${port}`);
});
