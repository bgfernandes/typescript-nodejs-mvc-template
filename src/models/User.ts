import { JSONSchema7 } from 'json-schema';
import BaseModel from './BaseModel';

export default class Users extends BaseModel {
  id: number | undefined;

  static get tableName():string {
    return 'users';
  }

  static get jsonSchema():JSONSchema7 {
    return {
      type: 'object',

      properties: {
        id: { type: 'integer' },
        created_at: { type: 'string' },
        updated_at: { type: 'string' }
      }
    };
  }
}
