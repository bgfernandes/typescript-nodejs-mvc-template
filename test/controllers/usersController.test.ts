import request from 'supertest';
import app from './../../src/app';
import User from './../../src/models/User';

describe('index', () => {
  it('renders the index page with no users', async () => {
    const response = await request(app).get('/users');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });

  it('renders the index page with some users', async () => {
    await User.query().insert({
      created_at: new Date('1995-12-17T03:24:00').toISOString(),
      updated_at: new Date('1995-12-17T03:24:00').toISOString(),
    });

    await User.query().insert({
      created_at: new Date('1997-12-17T03:24:00').toISOString(),
      updated_at: new Date('1997-12-17T03:24:00').toISOString(),
    });

    const response = await request(app).get('/users');
    expect(response.status).toBe(200);
    expect(response.text).toMatchSnapshot();
  });
});
