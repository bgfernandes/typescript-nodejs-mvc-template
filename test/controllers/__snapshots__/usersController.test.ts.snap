// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders the index page with no users 1`] = `
"
<!DOCTYPE html>
<html>
  <head>
    <title>Typescript Node MVC Template</title>

    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link href=\\"/tailwind.css\\" rel=\\"stylesheet\\">
  </head>

  <body>
    <header class=\\"flex flex-row justify-between border-b p-3\\">
  Typescript Node MVC Template

  <div>
    <a href=\\"/\\" class=\\"text-blue-500 capitalize\\"> home </a>
    |
    <a href=\\"/users/\\" class=\\"text-blue-500 capitalize\\"> users </a>
  </div>

  <div class=\\"relative\\">
    <input type=\\"checkbox\\" id=\\"language-selector\\" class=\\"hidden absolute\\">
    <label for=\\"language-selector\\" class=\\"flex items-center space-x-1 cursor-pointer\\">
      <span>en</span>
      <svg class=\\"h-4 w-4\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\" />
      </svg>
    </label>

    <div id=\\"languages-menu\\" class=\\"absolute mt-1 right-1 top-full min-w-max shadow rounded opacity-0 border border-gray-400 transition delay-75 ease-in-out z-10 bg-white\\">
      <ul class=\\"block text-right\\">
        
          <li><a href=\\"?lang=en\\" class=\\"block px-3 py-2\\">en</a></li>
        
          <li><a href=\\"?lang=pt-br\\" class=\\"block px-3 py-2\\">pt-br</a></li>
        
      </ul>
    </div>
  </div>
</header>


    <div class=\\"p-3\\">
     

<div class=\\"text-2xl capitalize\\"> users </div>

<table class=\\"border-collapse border\\">
  <thead>
    <tr>
      <th class=\\"border capitalize\\">id</th>
      <th class=\\"border capitalize\\">created at</th>
      <th class=\\"border capitalize\\">updated at</th>
    </tr>
  </thead>
  <tbody>
    
  </tbody>
</table>


    </div>
  </body>
</html>
"
`;

exports[`index renders the index page with some users 1`] = `
"
<!DOCTYPE html>
<html>
  <head>
    <title>Typescript Node MVC Template</title>

    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link href=\\"/tailwind.css\\" rel=\\"stylesheet\\">
  </head>

  <body>
    <header class=\\"flex flex-row justify-between border-b p-3\\">
  Typescript Node MVC Template

  <div>
    <a href=\\"/\\" class=\\"text-blue-500 capitalize\\"> home </a>
    |
    <a href=\\"/users/\\" class=\\"text-blue-500 capitalize\\"> users </a>
  </div>

  <div class=\\"relative\\">
    <input type=\\"checkbox\\" id=\\"language-selector\\" class=\\"hidden absolute\\">
    <label for=\\"language-selector\\" class=\\"flex items-center space-x-1 cursor-pointer\\">
      <span>en</span>
      <svg class=\\"h-4 w-4\\" xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" stroke=\\"currentColor\\">
        <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M19 9l-7 7-7-7\\" />
      </svg>
    </label>

    <div id=\\"languages-menu\\" class=\\"absolute mt-1 right-1 top-full min-w-max shadow rounded opacity-0 border border-gray-400 transition delay-75 ease-in-out z-10 bg-white\\">
      <ul class=\\"block text-right\\">
        
          <li><a href=\\"?lang=en\\" class=\\"block px-3 py-2\\">en</a></li>
        
          <li><a href=\\"?lang=pt-br\\" class=\\"block px-3 py-2\\">pt-br</a></li>
        
      </ul>
    </div>
  </div>
</header>


    <div class=\\"p-3\\">
     

<div class=\\"text-2xl capitalize\\"> users </div>

<table class=\\"border-collapse border\\">
  <thead>
    <tr>
      <th class=\\"border capitalize\\">id</th>
      <th class=\\"border capitalize\\">created at</th>
      <th class=\\"border capitalize\\">updated at</th>
    </tr>
  </thead>
  <tbody>
    
      <tr>
        <td class=\\"borde\\">1</td>
        <td class=\\"border\\">Sun Dec 17 1995 03:24:00 GMT+0000</td>
        <td class=\\"border\\">Sun Dec 17 1995 03:24:00 GMT+0000</td>
      </tr>
    
      <tr>
        <td class=\\"borde\\">2</td>
        <td class=\\"border\\">Wed Dec 17 1997 03:24:00 GMT+0000</td>
        <td class=\\"border\\">Wed Dec 17 1997 03:24:00 GMT+0000</td>
      </tr>
    
  </tbody>
</table>


    </div>
  </body>
</html>
"
`;
